src/zenml/orchestrators/step_runner.py

```diff
@@ -244,6 +244,11 @@ def run(
                     link_step_artifacts_to_model(
                         artifact_version_ids=output_artifact_ids
                     )
+                    # TODO: Why do we link the pipeline run to the model of the
```

Comment:
@avishniakov

Reply:
For this my current understanding is:
1. Output artifacts might specify a custom model in their `ArtifactConfig`, and we want to link those potential models to the pipeline run in which they were used
2. External artifacts can also reference a model to load a value: This seems to be a deprecated feature though, which was replaced by using either a model lazy loader or client lazy loader. If this is correct, can I also remove this as part of my deprecation cleanup?
3. Regular input artifacts can not specify a custom model, and the model which they're associated with is already linked to the pipeline run because they were an output of a previous step.
4. This is the missing piece here I think: When someone calls `load_artifact(...)` within a step, I think we do not link that artifact to the active model version. If you can confirm this and agree that we should add it, I'll add the logic to `load_artifact(...)`

Reply:
1. Yes
2. Can be removed
3. Yes
4. Yes, but it was intentionally made so. The thinking process is like the inputs are not really owned by a model and those links are more like an ownership thing. At least it was. If we agree it is logical to link all those artifacts - we can do it, no objections.

Reply:
Regarding 4.: I guess this is being consistent with how other step inputs work: None of them get explicitly linked to the model version.

Our model versions now track essentially the artifacts produced by the pipeline/step run that they're configured for. I'm still not entirely sure whether also linking all inputs makes sense, but for now I think we can keep it as is.